<launch>

  <arg name="gui" default="false" />

  <arg name="input_cloud" value="tf_cloud" />

  <node name="attention_clipper"
        pkg="nodelet" type="nodelet"
        args="standalone jsk_pcl/AttentionClipper">
    <remap from="~input/points" to="$(arg input_cloud)" />
    <rosparam>
      initial_pos: [0.25, 0, 0]
      initial_rot: [0, 0, 0]
      dimension_x: 0.1
      dimension_y: 0.2
      dimension_z: 0.1
      frame_id: base_link
    </rosparam>
  </node>

  <node name="extract_indices"
        pkg="jsk_pcl_ros" type="extract_indices">
    <remap from="~input" to="$(arg input_cloud)" />
    <remap from="~indices" to="tf_cloud_clipped" />
  </node>

	<!-- change your file path -->
	<arg name="file_name" value="$(find pcl_tools)/pcd/dobot.pcd"/>
	<!-- rate : if publish at 10 Hz, set 0.1 -->
	<arg name="rate" value="0.1"/>
	<node pkg="pcl_ros" type="pcd_to_pointcloud" name="pcd_to_pointcloud" args="$(arg file_name) $(arg rate)" output="screen">
		<remap from="cloud_pcd" to="dobot_points"/>
		<param name="frame_id" value="base_link" />
	</node>

</launch>

